aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,947 @@
+
+        assert!(exists<CollectionConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+        assert!(vector::length(&token_uris) == vector::length(&property_keys) && vector::length(&property_keys) == vector::length(&property_values) && vector::length(&property_values) == vector::length(&property_types), error::invalid_argument(EVECTOR_LENGTH_UNMATCHED));
+        let token_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+
+        assert!(vector::length(&token_uris) + vector::length(&token_config.token_uri) <= token_config.collection_maximum || token_config.collection_maximum == 0, error::invalid_argument(EEXCEEDS_COLLECTION_MAXIMUM));
+        vector::append(&mut token_config.token_uri, token_uris);
+        vector::append(&mut token_config.property_keys, property_keys);
+        vector::append(&mut token_config.property_values, property_values);
+        vector::append(&mut token_config.property_types, property_types);
+    }
+
+    /// Mint source certificate.
+    public entry fun mint_source_certificate(
+        nft_claimer: &signer,
+        amount: u64
+    ) acquires NFTMintConfig, PublicMintConfig, WhitelistMintConfig, CollectionConfig, SourceToken {
+        assert!(exists<CollectionConfig>(@post_mint_reveal_nft) && exists<WhitelistMintConfig>(@post_mint_reveal_nft) && exists<PublicMintConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(@post_mint_reveal_nft);
+        let public_mint_config = borrow_global<PublicMintConfig>(@post_mint_reveal_nft);
+
+        let now = timestamp::now_seconds();
+        assert!((now > whitelist_mint_config.whitelist_minting_start_time && now < whitelist_mint_config.whitelist_minting_end_time) || (now > public_mint_config.public_minting_start_time && now < public_mint_config.public_minting_end_time), error::permission_denied(EMINTING_IS_NOT_ENABLED));
+
+        let nft_mint_config = borrow_global<NFTMintConfig>(@post_mint_reveal_nft);
+        let collection_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+        let resource_signer = create_signer_with_capability(&nft_mint_config.signer_cap);
+
+        // Set up a source token if it doesn't exist yet.
+        if (!exists<SourceToken>(@post_mint_reveal_nft)) {
+            // This is the collection that contains certificate tokens, which will allow users to exchange to an NFT in the destination collection after reveal time.
+            let source_collection_name = collection_config.collection_name;
+            string::append_utf8(&mut source_collection_name, b" - source collection");
+            create_collection(&resource_signer, source_collection_name, collection_config.collection_description, utf8(b""), collection_config.collection_maximum, collection_config.collection_mutate_config);
+
+            let source_token_data_id = create_tokendata(
```

Comment:
how about init a fungible token collection in the init_module? Here, we just provide the minter a fungible token to further reduce gas cost?

Reply:
oh yeah, so I think we're minting non-fungible tokens and mutating the token properties because we care about the `property_version` - it'll be added to the token name when they exchange for a destination token. otherwise minting earlier or later makes no difference on the name of the destination token. (I guess most people would prefer to have a small number on the token name? that's why people rush to mint when a collection starts minting right?)

Reply:
ok, make sense. sounds good.